src/main/java/ch/njol/skript/registrations/Arithmetics.java

```diff
@@ -0,0 +1,140 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.registrations;
+
+import ch.njol.skript.expressions.arithmetic.Operator;
+import com.google.common.base.Preconditions;
+import org.checkerframework.checker.units.qual.A;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.lang.arithmetic.Arithmetic;
+import org.skriptlang.skript.lang.arithmetic.Arithmetic.ArithmeticInfo;
+import org.skriptlang.skript.lang.arithmetic.Difference;
+import org.skriptlang.skript.lang.arithmetic.Difference.DifferenceInfo;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * Contains all registered arithmetics and allows operating with them.
+ */
+@SuppressWarnings("unchecked")
+public abstract class Arithmetics {
+
+	private Arithmetics() {}
+
+	private static final List<ArithmeticInfo<?>> registeredArithmetics = new ArrayList<>();
+	private static final List<DifferenceInfo<?, ?>> registeredDifferences = new ArrayList<>();
+
+	public static <T> void registerArithmetic(Class<T> type, Arithmetic<T> arithmetic) {
+		Preconditions.checkArgument(getArithmeticInfoExact(type) == null, "An arithmetic is already registered with this type");
+		registeredArithmetics.add(new ArithmeticInfo<>(type, arithmetic));
+	}
+
+	@Nullable
+	public static <T> Arithmetic<? super T> getArithmetic(Class<T> type) {
+		ArithmeticInfo<? super T> arithmeticInfo = getArithmeticInfo(type);
+		if (arithmeticInfo == null)
+			return null;
+		return arithmeticInfo.getArithmetic();
+	}
+
+	@Nullable
+	public static <T> ArithmeticInfo<? super T> getArithmeticInfo(Class<T> type) {
```

Comment:
If these getInfo methods are to remain public, they Info classes themselves should also be public

Tbh I think it would be a good idea to extract the info classes into the main package (instead of keeping them within Arithmetic/Difference)

Reply:
They are public, anything declared inside an interface is automatically implicitly public.

Reply:
Of course - silly me!

I think the extraction is still a good idea :D